src/Fabulous.AST/Widgets/TypeDefinitions/Union.fs

```diff
@@ -34,20 +36,37 @@ module Union =
 
             let multipleAttributes =
                 match attributes with
-                | ValueSome values -> TypeHelpers.createAttributes values |> Some
+                | ValueSome values -> MultipleAttributeListNode.Create values |> Some
+                | ValueNone -> None
+
+            let typeParams = Helpers.tryGetScalarValue widget TypeParams
+
+            let typeParams =
+                match typeParams with
+                | ValueSome values ->
+                    TyparDeclsPostfixListNode(
+                        SingleTextNode.lessThan,
+                        [ for v in values do
+                              TyparDeclNode(None, SingleTextNode.Create v, Range.Zero) ],
+                        [],
+                        SingleTextNode.greaterThan,
+                        Range.Zero
+                    )
+                    |> TyparDecls.PostfixList
+                    |> Some
```

Comment:
I think we could pass the `typeParams` as optional in the widget constructor. As this will imply is belongs to the type and will avoid having to wrap the union witn extra `()`
```fsharp
Union("Colors", ["'other"; ""]) {
    UnionParameterizedCase("Red") {
        Field("a", Type.FromString "string")
        Field("b", Type.FromString "'other")
    }

    UnionCase("Green")
    UnionCase("Blue")
    UnionCase("Yellow")
}
```

Reply:
I don't think I really have any preference.
What is the typical heuristic to favor `.someOtherthing() {` over an overload?